### PR TITLE
Upgrade Python versions

### PR DESCRIPTION
**Pull request by ulgens**
Upgrade Python versions:
- 3.8.2 to 3.8.3
- 2.7.17 to 2.8.18

### CODE REVIEW

**Comment by ulgens** [comment]
`Line is too long. [81/80]` 🙃

**Comment by ulgens** [comment]
ping @jurre @feelepxyz

File: Dockerfile [hunk 1]
```diff
@@ -67,10 +67,10 @@ RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys C3173AA6 \
 ENV PYENV_ROOT=/usr/local/.pyenv \
     PATH="/usr/local/.pyenv/bin:$PATH"
 RUN git clone https://github.com/pyenv/pyenv.git /usr/local/.pyenv \
-    && cd /usr/local/.pyenv && git checkout v1.2.17 && cd - \
```
Comment (thepwagner) [inline Dockerfile:70]: I'm concerned about accepting all updates that flow into the `master` branch upstream.

Can we wait for the next release?
If not, could we pin to a specific commit?

File: python/spec/dependabot/python/update_checker/pipenv_version_resolver_spec.rb [hunk 2]
```diff
@@ -255,7 +255,7 @@
                 to start_with("Dependabot detected the following Python")
               expect(error.message).to include("3.4.*")
               expect(error.message).
-                to include("supported in Dependabot: 3.8.2, 3.8.1, 3.8.0")
+                to include("supported in Dependabot: 3.8.3, 3.8.2, 3.8.1, 3.8.0")
```
Comment (thepwagner) [inline python/spec/dependabot/python/update_checker/pipenv_version_resolver_spec.rb:258]:
```suggestion
                to include("supported in Dependabot: 3.8.3, 3.8.2, 3.8.1")
```

Weakens the assertion slightly, but satisfies the linter.

File: Dockerfile (hunk 1 above)
Comment (ulgens) [reply Dockerfile:70]: @thepwagner Fixed.

**Comment by jurre** [comment]
Seems like an unrelated failure in the `hex` module, will try looking into this today.

**Comment by jurre** [comment]
Could you try rebasing on / merging in master @ulgens? Tests should pass after that

**Comment by ulgens** [comment]
@jurre Done but still failing. It seems it's bundler this time.

**Comment by jurre** [comment]
I've rerun the tests 🤞

**Comment by jurre** [comment]
I figured out why it broke, will fix on master and then merge this, thanks @ulgens !

**Pull request merged**